python-threatexchange/threatexchange/exchanges/helpers.py

```diff
@@ -145,3 +148,91 @@ def get_for_signal_type(
                 if by_signal:
                     ret[collab.name] = by_signal
         return ret
+
+    def exists(self, collab: CollaborationConfigBase) -> bool:
+        raise NotImplementedError
+
+
+class DMBFetchedStateStore(fetch_state.FetchedStateStoreBase):
+    def __init__(
+        self,
+        api_cls: TSignalExchangeAPICls,
+    ) -> None:
+        self.api_cls = api_cls
+
+
+    def get_for_signal_type(
+        self, collabs: t.List[CollaborationConfigBase], signal_type: t.Type[SignalType]
+    ) -> t.Dict[str, t.Dict[str, fetch_state.FetchedSignalMetadata]]:
+        """
+        Get as a map of CollabConfigBase.name() => {signal: Metadata}
+
+        This is meant for simple storage and indexing solutions, but at
+        scale, you likely want to store as IDs rather than the full metadata.
+
+        TODO: This currently implies that you are going to load the entire dataset
+        into memory, which once we start getting huge amounts of data, might not make
+        sense.
+        """
+
+        ret = {}
+        for collab in collabs:
+            with dbm.open(collab.name, 'c') as db:
+                k = db.firstkey()
+                data = {}
+                while k:
+                    print(k, db[k])
+                    _k, _v = map(pickle.loads, [k, db[k]])
```

Comment:
nit: underscores usually denote that the variable is unused (not that it's private as in other language conventions), so `k`, `v` is probably preferred.